x/sync/network_server.go

```diff
@@ -215,10 +231,12 @@ func (s *NetworkServer) HandleChangeProofRequest(
 		}
 
 		if len(proofBytes) < bytesLimit {
+			// TODO handle this fatal error
 			return s.appSender.SendAppResponse(ctx, nodeID, requestID, proofBytes)
 		}
-		// the proof size was too large, try to shrink it
-		keyLimit = uint32(len(changeProof.KeyChanges)) / 2
+
+		// The proof was too large. Try to shrink it.
+		keyLimit /= 2
```

Comment:
This seems to have been changed from `keyLimit = uint32(len(changeProof.KeyChanges)) / 2` - I think the prior code made more sense no?

Reply:
changed